backend/src/src-moderation/src/main/java/org/eclipse/sw360/moderation/db/ModerationRequestGenerator.java

```diff
@@ -181,6 +181,37 @@ protected void dealWithCustomMap(U field) {
         documentDeletions.setFieldValue(field, deleteMap);
     }
 
+    protected void dealWithStringtoStringMap(U field) {
+        Map<String, String> updateDocumentMap = CommonUtils.nullToEmptyMap(
+                (Map<String, String>) updateDocument.getFieldValue(field));
+        Map<String, String> actualDocumentMap = CommonUtils.nullToEmptyMap(
+                (Map<String, String>) actualDocument.getFieldValue(field));
+        if (updateDocumentMap.equals(actualDocumentMap)) {
+            return;
+        }
+
+        Map<String,String> addedMap = updateDocumentMap;
+        Map<String, String> deletedMap = actualDocumentMap;
+
+        Set<String> commonKeys = Sets.intersection(updateDocumentMap.keySet(), actualDocumentMap.keySet());
+        for(String id : commonKeys) {
+            String actual = actualDocumentMap.get(id);
+            String update = updateDocumentMap.get(id);
+            try {
+                if(! actual.equals(update)) {
+                    addedMap.put(id, update);
+                    deletedMap.put(id, actual);
+                }
+            } catch (NullPointerException e) {
```

Comment:
This is exactly the wrong solution. As a general rule, you don't ever catch a NullPointerException. You check the reference for null before dereferencing it.
In this specific case, comparing with `Objects.equals()` would be a sufficient and safe alternative, because it already handles null values correctly.

Reply:
Hah, OK. Good to know for future :+1: